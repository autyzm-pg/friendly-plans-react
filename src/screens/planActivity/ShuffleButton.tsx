import React, { SFC } from 'react';
import { StyleSheet } from 'react-native';

import { i18n } from 'locale';
import { FlatButton } from '../../components';
import { dimensions, palette } from '../../styles';

interface Props {
  disabled?: boolean;
}

export const ShuffleButton: SFC<Props> = ({ disabled }) => (
  <FlatButton
    title={i18n.t('planActivity:shuffleTasks')}
    icon={{
      name: 'shuffle',
      type: 'material-community-icons',
      color: disabled ? palette.textDisabled : palette.primary,
      size: 22,
      disabled,
      disabledStyle: styles.iconDisabled,
    }}
    disabled={disabled}
    buttonStyle={styles.button}
  />
);

const styles = StyleSheet.create({
  button: {
    marginRight: dimensions.spacingBig,
  },
  iconDisabled: {
    backgroundColor: 'transparent',
  },
});
