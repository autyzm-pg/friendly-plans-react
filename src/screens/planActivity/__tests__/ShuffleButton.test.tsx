import { shallow } from 'enzyme';
import React from 'react';
import { ShuffleButton } from '../ShuffleButton';

describe('ShuffleButton', () => {
  it('renders correctly', () => {
    const shuffleButton = shallow(<ShuffleButton />);

    expect(shuffleButton).toMatchSnapshot();
  });
});
