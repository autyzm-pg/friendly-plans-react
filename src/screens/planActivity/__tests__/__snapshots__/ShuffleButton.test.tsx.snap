// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShuffleButton renders correctly 1`] = `
<FlatButton
  buttonStyle={
    Object {
      "marginRight": 24,
    }
  }
  icon={
    Object {
      "color": "#6654EB",
      "disabled": undefined,
      "disabledStyle": Object {
        "backgroundColor": "transparent",
      },
      "name": "shuffle",
      "size": 22,
      "type": "material-community-icons",
    }
  }
  title="Shuffle tasks"
/>
`;
